src/k8s-engine/lima.ts

```diff
@@ -1538,23 +1608,29 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
     // If we're in the middle of starting, also ignore the call to stop (from
     // the process terminating), as we do not want to shut down the VM in that
     // case.
+
     if (this.currentAction !== Action.NONE) {
       return;
     }
     this.currentAction = Action.STOPPING;
 
-    this.lastCommandComment = 'Stopping Kubernetes';
+    this.lastCommandComment = 'Stopping services';
     await this.progressTracker.action(this.lastCommandComment, 10, async() => {
       try {
         this.setState(K8s.State.STOPPING);
 
         const status = await this.status;
 
         if (defined(status) && status.status === 'Running') {
-          await this.ssh('sudo', '/sbin/rc-service', 'k3s', 'stop');
-          await this.ssh('sudo', '/sbin/rc-service', '--ifstarted', 'docker', 'stop');
-          // Always stop it, even if we're on MOBY, in case it got started for some reason.
-          await this.ssh('sudo', '/sbin/rc-service', '--ifstarted', 'buildkitd', 'stop');
+          if (this.#enabledK3s) {
+            await this.ssh('sudo', '/sbin/rc-service', 'k3s', 'stop');
+            // Always stop it, even if we're on MOBY, in case it got started for some reason.
+            await this.ssh('sudo', '/sbin/rc-service', '--ifstarted', 'buildkitd', 'stop');
+          } else if (this.#currentContainerEngine === ContainerEngine.MOBY) {
+            await this.ssh('sudo', '/sbin/rc-service', '--ifstarted', 'docker', 'stop');
+          } else {
+            await this.ssh('sudo', '/sbin/rc-service', '--ifstarted', 'containerd', 'stop');
+          }
```

Comment:
Just stop both.

Reply:
Yes, that works because these services are always known.